pkgs/by-name/xe/xephem/package.nix

```diff
@@ -0,0 +1,83 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeDesktopItem,
+  copyDesktopItems,
+  installShellFiles,
+  motif,
+  openssl,
+  groff,
+  xorg,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xephen";
+  version = "4.2.0";
+
+  buildInputs = [
+    motif
+    openssl
+    groff
+    xorg.libXmu
+    xorg.libXext
+    xorg.libXt
+  ];
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    installShellFiles
+  ];
+
+  src = fetchFromGitHub {
+    owner = "XEphem";
+    repo = "XEphem";
+    rev = version;
+    hash = "sha256-TuzXrWoJOAHg31DrJObPcHBXgtqR/KWKFRsqddPzL4c=";
+  };
+
+  sourceRoot = "${src.name}/GUI/xephem";
```

Comment:
Seems to cause errors when building
```
Assembler messages:
Fatal error: can't create aa_hadec.o: Permission denied
```

Reply:
`sourceRoot` is not a formal parameter. It is set as environment variable and used in `pkgs/stdenv/generic/setup.sh` (line 1300+).

Probably the problem is that the mode bit of parent directories is not changed to allow writing, when using sourceRoot.